### PR TITLE
[Core] Update string utilities adding ReplaceAllSubstrings

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
Update string utilities adding ReplaceAllSubstrings. Part of #7899

**Changelog**
- Update string utilities adding ReplaceAllSubstrings

### CODE REVIEW

File: kratos/tests/cpp_tests/utilities/test_string_utilities.cpp [hunk 1]
```diff
@@ -69,5 +69,13 @@ KRATOS_TEST_CASE_IN_SUITE(SplitStringByDelimiter, KratosCoreFastSuite)
     KRATOS_CHECK_STRING_EQUAL(splitted_string[3], "   ");
 }
 
+KRATOS_TEST_CASE_IN_SUITE(ReplaceAllSubstrings, KratosCoreFastSuite)
+{
+    const std::string string_to_replace = "Pokemon is an awesome show. Pokemon is the best!";
+    const std::string correct_string = StringUtilities::ReplaceAllSubstrings(string_to_replace, "Pokemon", "JoJo Bizarre Addenture");
+
+    KRATOS_CHECK_STRING_EQUAL(correct_string, "JoJo Bizarre Addenture is an awesome show. JoJo Bizarre Addenture is the best!");
```
Comment (loumalouomega) [inline kratos/tests/cpp_tests/utilities/test_string_utilities.cpp]: @roigcarlo 

imagen

File: kratos/utilities/string_utilities.h [hunk 2]
```diff
@@ -96,6 +96,13 @@ namespace StringUtilities
         const std::string& rString,
         const char Delimiter);
 
+    /**
+     * @brief This function replaces from a string all times a certain substring is repeated
+     * @param from The original string to be replaced
+     * @param to The string which replaces the substring
+     */
+    std::string KRATOS_API(KRATOS_CORE) ReplaceAllSubstrings(std::string str, const std::string& from, const std::string& to);
```
Comment (philbucher) [inline kratos/utilities/string_utilities.h]:
```suggestion
    std::string KRATOS_API(KRATOS_CORE) ReplaceAllSubstrings(const std::string& rStringToBeReplaced, str, const std::string& rOldSubstring, const std::string& rNewSubstring);
```

seeing `std::string` being passed by value makes me cringe xD

also please name the arguments better, I cannot understand without looking at the code what is what (I only made a suggestion but pretty sure I got it wrong)

Comment (loumalouomega) [reply kratos/utilities/string_utilities.h]: Sorry, it was a function copy pasted from stackoverflow, one sec

**Review by philbucher — DISMISSED** [review]

**Review by philbucher — DISMISSED** [review]

**Comment by loumalouomega** [comment]
@philbucher re-approval, please

**Review by philbucher — APPROVED** [review]

**Pull request merged**